pkg/login/social/connectors/google_oauth.go

```diff
@@ -298,11 +298,13 @@ func (s *SocialGoogle) getGroupsPage(ctx context.Context, client *http.Client, u
 }
 
 func (s *SocialGoogle) isHDAllowed(hd string) error {
-	if len(s.info.AllowedDomains) == 0 {
+	info := s.GetOAuthInfo()
```

Comment:
won't this cause 2 separate configs to get mixed if half the flow is made with a config and the other half with another?

Reply:
I don't think we can have mixed params in the `info` struct because every time we reload it a new info struct is being created and the new one replaces the old one. Also the reads/writes from/to `info` are protected by a mutex so I think it's fine.